### DISCUSSION TITLE
allow exception (for admin) on sidebar_editable: false

### DISCUSSION THREAD

**Discussion by Mariusthvdb** [elchininet/custom-sidebar#66] [Ideas]
just learned about the sidebar_editable:
```
{
    "title": "Rumah kami",
    "sidebar_editable": false,
    "order": [
```

which is really useful.

I've tried setting an exception to that, but either made a mistake, or not possible, so would like to ask if you would consider adding that option.

btw, the above setting does Not make the glowing circle go away when holding the title. 

<img width="234" alt="Scherm­afbeelding 2024-03-05 om 11 59 56" src="https://github.com/elchininet/custom-sidebar/assets/33354141/b1938afe-5108-4f49-b531-fb31f2c0f06c">

Maybe that could be also prevented incase of editable: false?

thanks!

**Comment by elchininet** [comment]
>I've tried setting an exception to that, but either made a mistake, or not possible, so would like to ask if you would consider adding that option.

You didn‘t make a mistake, exceptions are only intended to override the `order`, the rest of the options (`title`, `sidebar_editable` and `styles`) are global. It is something that could be achieved but not without a whole refactor because since the beginning the `exceptions` are intended to override just the order. So, it is not something that will be done any time soon.

>Maybe that could be also prevented incase of editable: false?

Hard to achieve it without disabling the sidebar button because it is triggered from the menu element (the one that contains the button and the title). I need to dig a bit in the Home Assistant source code to see if it can be prevented in some way.

**Reply by Mariusthvdb** [reply]
understand, ofc I also looked for that in the core Frontend code, but wasnt able to pinpoint it yet. Ill dig with you ;-)

**Reply by elchininet** [reply]
[This](https://github.com/home-assistant/frontend/blob/32e8d2043c3e3595880eea9d007db38768545908/src/components/ha-sidebar.ts#L343-L345) is the culprit. It renders [this element](https://github.com/home-assistant/frontend/blob/32e8d2043c3e3595880eea9d007db38768545908/src/panels/lovelace/common/directives/action-handler-directive.ts#L41) in the root when one presses on the sidebar header (and also on [other elements](https://github.com/search?q=repo%3Ahome-assistant%2Ffrontend+action-handler&type=code&p=1)). Didn‘t find how to disable it only for the sidebar though.

**Comment by elchininet** [comment]
Hi @Mariusthvdb,
Please, read the notes of the last release. Now it is possible to put `sidebar_editable` in an `exception` and it also tries to hide the glowing circle when you click and hold.
Regards

**Reply by Mariusthvdb** [reply]
seems to work! Nice, thanks.

I now need to read up in the docs, to see if I can template that functionality.
As I've posted in the other discussion, I used this for now:
```
      .menu {
        {{'pointer-events: none' if is_state('input_boolean.hide_menubutton','on')}};
      }
```
in card-mod-themes, and have grown attached to that option of setting it based on the input_boolean. So would love to get that into the `sidebar_editable` exception ;-)

___edit___

so I added this:
```
sidebar_editable: >
  {{is_state('input_boolean.hide_menubutton','off')}}
```

and it works beautifully!
I did have to redo the complete order, but after that it seems to be as hoped for

**Reply by elchininet** [reply]
>I now need to read up in the docs, to see if I can template that functionality.

Yes you can, it is one of the new features.

>and it works beautifully!
>I did have to redo the complete order, but after that it seems to be as hoped for

Glad that it is working, thanks for confirming 👍
